### PR TITLE
Add QuadrumLabs Delta

### PR DESCRIPTION
**Pull request by Sleepdealr**
This replaces #18198 because lfg went AWOL

I had to remake this from the gerbers and it was awful

## Description

Adds QuadrumLabs Delta to QMK

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] Enhancement/optimization
- [ ] New feature
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/quadrum/delta/rules.mk [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# This file intentionally left blank
+
+ENCODER_ENABLE = yes
+ENCODER_MAP_ENABLE = yes
```
Comment (zvecr) [inline keyboards/quadrum/delta/rules.mk:4]: `ENCODER_MAP_ENABLE` should never be enabled at the keyboard level
```suggestion
```

File: keyboards/quadrum/delta/config.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// Copyright 2023 Sleepdealer (@Sleepdealr)
```
Comment (zvecr) [inline keyboards/quadrum/delta/config.h:1]: This file should be removed

File: keyboards/quadrum/delta/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+     KC_TRNS,  KC_TRNS,  KC_TRNS,                                          KC_TRNS,                      KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS
+     )
+
+};
+
+const uint16_t PROGMEM encoder_map[][1][2] = {
+    [0] =  { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [1] =  { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [2] =  { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) },
+    [3] =  { ENCODER_CCW_CW(KC_VOLD, KC_VOLU) }
```
Comment (zvecr) [inline keyboards/quadrum/delta/keymaps/default/keymap.c:57]:
```suggestion
```

File: keyboards/quadrum/delta/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+     KC_TRNS,  KC_TRNS,  KC_TRNS,                                          KC_TRNS,                      KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS
+     ),
+     [3] = LAYOUT(
+     KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS,
+     KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS,
+     KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS,
+     KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS,
+     KC_TRNS,  KC_TRNS,  KC_TRNS,                                          KC_TRNS,                      KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS
+     )
+
```
Comment (zvecr) [inline keyboards/quadrum/delta/keymaps/default/keymap.c:50]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]
`keyboards/quadrum/delta/keymaps/default/rules.mk` should be added with just `ENCODER_MAP_ENABLE = yes`

**Comment by Sleepdealr** [comment]
How can I remove the lib stuff from it? Every time I commit it seems to include it for whatever reason

**Comment by Sleepdealr** [comment]
Fixed, thanks @waffle87!

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by sigprof — CHANGES_REQUESTED** [review]
Layouts need some work — apparently what you have here is `LAYOUT_all` (unbuildable as it is, because ANSI backslash is separate from ISO NUHS on this PCB, therefore you need to use 1.25u Enter to make all matrix positions assignable), but a lone `LAYOUT_all` is not allowed by the rules, so you would need to add at least `LAYOUT_ansi` and `LAYOUT_iso` with appropriate modifications.  (Maybe you should also provide layout variations for 2u Backspace, but you would need to check which matrix location corresponds to the 2u position.)

File: keyboards/quadrum/delta/info.json [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+                { "matrix": [2, 4], "x": 4.75, "y": 2.0 },
+                { "matrix": [2, 5], "x": 5.75, "y": 2.0 },
+                { "matrix": [2, 6], "x": 6.75, "y": 2.0 },
+                { "matrix": [2, 7], "x": 7.75, "y": 2.0 },
+                { "matrix": [2, 8], "x": 8.75, "y": 2.0 },
+                { "matrix": [2, 9], "x": 9.75, "y": 2.0 },
+                { "matrix": [2, 10], "x": 10.75, "y": 2.0 },
+                { "matrix": [2, 11], "x": 11.75, "y": 2.0 },
+                { "matrix": [2, 12], "x": 12.75, "y": 2.0 },
+                { "matrix": [2, 13], "w": 2.25, "x": 12.75, "y": 2.0 },
```
[the remark this answers is not in the document]
Comment (Sleepdealr) [reply keyboards/quadrum/delta/info.json:83]: Just to clarify
After adding this in order to make it not overlap in the configurator, are separate layouts required? Or just a suggestion?
Most people will end up using via anyway.

File: keyboards/quadrum/delta/keymaps/ansi/keymap.c [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+     KC_GRAVE, KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,  KC_F12,           QK_RBT,   KC_TRNS,
+     KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS,
+     KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,           KC_TRNS,          KC_TRNS,  KC_TRNS,
+     KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS,
+     KC_TRNS,  KC_TRNS,  KC_TRNS,                                          KC_TRNS,                      KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS
+     )
+};
+
+#ifdef ENCODER_MAP_ENABLE
+const uint16_t PROGMEM encoder_map[][1][2] = {
```
Comment (zvecr) [inline keyboards/quadrum/delta/keymaps/ansi/keymap.c:38]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```

File: keyboards/quadrum/delta/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+     KC_GRAVE, KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,  KC_F12,           QK_RBT,   KC_TRNS,
+     KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS,
+     KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS,
+     KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS,
+     KC_TRNS,  KC_TRNS,  KC_TRNS,                                          KC_TRNS,                      KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS
+     )
+};
+
+#ifdef ENCODER_MAP_ENABLE
+const uint16_t PROGMEM encoder_map[][1][2] = {
```
Comment (zvecr) [inline keyboards/quadrum/delta/keymaps/default/keymap.c:38]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```

File: keyboards/quadrum/delta/keymaps/iso/keymap.c [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+     KC_GRAVE, KC_F1,    KC_F2,    KC_F3,    KC_F4,    KC_F5,    KC_F6,    KC_F7,    KC_F8,    KC_F9,    KC_F10,   KC_F11,  KC_F12,           QK_RBT,   KC_TRNS,
+     KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,           KC_TRNS,          KC_TRNS,  KC_TRNS,
+     KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS,
+     KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS,
+     KC_TRNS,  KC_TRNS,  KC_TRNS,                                          KC_TRNS,                      KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS
+     )
+};
+
+#ifdef ENCODER_MAP_ENABLE
+const uint16_t PROGMEM encoder_map[][1][2] = {
```
Comment (zvecr) [inline keyboards/quadrum/delta/keymaps/iso/keymap.c:38]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```

File: keyboards/quadrum/delta/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+     KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS,
+     KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS,
+     KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS,  KC_TRNS, KC_TRNS,          KC_TRNS,  KC_TRNS,
+     KC_TRNS,  KC_TRNS,  KC_TRNS,                                          KC_TRNS,                      KC_TRNS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS
+     )
+
+};
+
+#ifdef ENCODER_MAP_ENABLE
+const uint16_t PROGMEM encoder_map[][1][2] = {
```
Comment (zvecr) [inline keyboards/quadrum/delta/keymaps/via/keymap.c:57]:
```suggestion
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][2] = {
```

File: keyboards/quadrum/delta/keymaps/ansi/keymap.c [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+*/
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_ansi(
+        KC_ESC,  KC_1,     KC_2,   KC_3,   KC_4,   KC_5,   KC_6,   KC_7,   KC_8,    KC_9,     KC_0,     KC_MINS,  KC_EQL,   KC_BSPC,  KC_DEL,   KC_MUTE,
+        KC_TAB,  KC_Q,     KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,    KC_O,     KC_P,     KC_LBRC,  KC_RBRC,  KC_BSLS,  KC_PGUP,
+        KC_CAPS, KC_A,     KC_S,   KC_D,   KC_F,   KC_G,   KC_H,   KC_J,   KC_K,    KC_L,     KC_SCLN,  KC_QUOT,            KC_ENT,   KC_PGDN,
+        KC_LSFT, KC_NUBS,  KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,   KC_N,   KC_M,    KC_COMM,  KC_DOT,   KC_SLSH,  KC_RSFT,            KC_UP,
```
Comment (zvecr) [inline keyboards/quadrum/delta/keymaps/ansi/keymap.c:25]: `KC_LSFT, KC_NUBS,  KC_Z` would highlight that the ansi layout macro is wrong, as `KC_NUBS` would generally not feature on an ansi board.

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/quadrum/delta/keymaps/ansi/keymap.c (hunk 10 above)
Comment (Sleepdealr) [reply keyboards/quadrum/delta/keymaps/ansi/keymap.c:25]: Why not just leave it in?
Can let the person decide if they want to use split LShift or not, even if it's usually not on an ANSI enter board.

Comment (zvecr) [reply keyboards/quadrum/delta/keymaps/ansi/keymap.c:25]: Because that is the general convention followed.

Comment (Sleepdealr) [reply keyboards/quadrum/delta/keymaps/ansi/keymap.c]: Sure I guess
Changed

File: keyboards/quadrum/delta/info.json [hunk 11]
```diff
@@ -0,0 +1,257 @@
...
+            {
+                "pin_a": "D0",
+                "pin_b": "D1"
+            }
+        ]
+    },
+    "layouts": {
+        "LAYOUT_all": {
+            "layout": [
+                { "matrix": [0, 0], "x": 0.0, "y": 0.0 },
```
Comment (fauxpark) [inline keyboards/quadrum/delta/info.json:39]:
```suggestion
                { "matrix": [0, 0], "x": 0, "y": 0 },
```
etc.

**Review by drashna — APPROVED** [review]

**Review by nooges — APPROVED** [review]

**Review by keyboard-magpie — APPROVED** [review]

**Pull request merged**